src/components/MAPI/clusters/utils.ts

```diff
@@ -0,0 +1,149 @@
+import ErrorReporter from 'lib/errors/ErrorReporter';
+import { IRelease, ReleaseHelper } from 'lib/ReleaseHelper';
+import { IMachineType } from 'MAPI/utils';
+import * as capiv1alpha3 from 'model/services/mapi/capiv1alpha3';
+import * as capiexpv1alpha3 from 'model/services/mapi/capiv1alpha3/exp';
+import * as capzexpv1alpha3 from 'model/services/mapi/capzv1alpha3/exp';
+import * as releasev1alpha1 from 'model/services/mapi/releasev1alpha1';
+import { Providers } from 'shared/constants';
+import { PropertiesOf } from 'shared/types';
+
+export function getWorkerNodesCount(
+  nodePools?: capiv1alpha3.IMachineDeployment[] | capiexpv1alpha3.IMachinePool[]
+) {
+  if (!nodePools) return undefined;
+
+  let count = 0;
+  for (const nodePool of nodePools) {
+    if (typeof nodePool.status?.readyReplicas !== 'undefined') {
+      count += nodePool.status.readyReplicas;
+    }
+  }
+
+  return count;
+}
+
+export function getWorkerNodesCPU(
+  nodePools?:
+    | capiv1alpha3.IMachineDeployment[]
+    | capiexpv1alpha3.IMachinePool[],
+  providerNodePools?: capzexpv1alpha3.IAzureMachinePool[],
+  machineTypes?: Record<string, IMachineType>
+) {
+  if (!nodePools || !providerNodePools || !machineTypes) return undefined;
+
+  let count = 0;
+
+  for (let i = 0; i < providerNodePools.length; i++) {
+    const vmSize = providerNodePools[i].spec?.template.vmSize;
+    const readyReplicas = nodePools[i].status?.readyReplicas;
```

Comment:
Since provider node pools are fetched based on existing node pools, the arrays have the same length and are ordered in the same way